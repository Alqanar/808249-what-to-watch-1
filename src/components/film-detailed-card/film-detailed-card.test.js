import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {createStore} from "redux";

import FilmDetailedCard from "./film-detailed-card.tsx";

const movieMock = {
  backgroundColor: `#ffffff`,
  coverLink: `https://m.media-amazon.com/images/M/MV5BNmE1ZGI2ZDctZDQxNC00MmRkLWJkYTQtNmY3YmI5MzlhOTVkXkEyXkFqcGdeQXVyMjc1NDA2OA@@._V1_SY1000_CR0,0,1498,1000_AL_.jpg`,
  description: `Post World War II, a British colonel and his wife are assigned to live in Hamburg during the post-war reconstruction, but tensions arise with the German who previously owned the house.`,
  director: `James Kent`,
  genre: [`Drama`, `Romance`, `War`],
  id: `0`,
  favorite: false,
  name: `The Aftermath`,
  posterImage: `https://m.media-amazon.com/images/M/MV5BMTk2MDEyNTE5M15BMl5BanBnXkFtZTgwMzY1NDM4NjM@._V1_SY1000_CR0,0,674,1000_AL_.jpg`,
  posterLink: `https://st.kp.yandex.net/images/film_iphone/iphone360_1000125.jpg`,
  trailer: `https://youtu.be/FPv3e2FZOgo`,
  rating: 6.3,
  released: 2019,
  duration: 108,
  scoresCount: 2726,
  famousActors: [`Keira Knightley`, `Ned Wills`, `Pandora Colin`],
  videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`
};

const testInitialState = {
  genre: `All genres`,
  films: [],
  authorization: {
    user: {
      id: `1`
    }
  }
};

describe(`Featured film card correctly renders`, () => {
  it(`when needVanish not passed renders div wrapper`, () => {
    const tree = renderer
      .create(
        <Provider store={createStore(() => testInitialState)}>
          <FilmDetailedCard
            film={movieMock}
            className={`movie-card__info`}
          />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`when needVanish not passed renders without wrapper`, () => {
    const tree = renderer
      .create(
        <Provider store={createStore(() => testInitialState)}>
          <FilmDetailedCard
            film={movieMock}
            needVanish
          />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
