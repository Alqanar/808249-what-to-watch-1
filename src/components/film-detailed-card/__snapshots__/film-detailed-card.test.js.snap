// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Featured film card correctly renders when needVanish not passed renders div wrapper 1`] = `
<div
  className="movie-card__info"
>
  <div
    className="movie-card__poster "
  >
    <img
      alt="The Aftermath poster"
      height="327"
      src="https://m.media-amazon.com/images/M/MV5BMTk2MDEyNTE5M15BMl5BanBnXkFtZTgwMzY1NDM4NjM@._V1_SY1000_CR0,0,674,1000_AL_.jpg"
      width="218"
    />
  </div>
  <div
    className="movie-card__desc"
  >
    <h2
      className="movie-card__title"
    >
      The Aftermath
    </h2>
    <p
      className="movie-card__meta"
    >
      <span
        className="movie-card__genre"
      >
        Drama, Romance, War
      </span>
      <span
        className="movie-card__year"
      >
        2019
      </span>
    </p>
    <div
      className="movie-card__buttons"
    >
      <button
        className="btn btn--play movie-card__button"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <button
        className="btn btn--list movie-card__button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          height="20"
          viewBox="0 0 19 20"
          width="19"
        >
          <use
            xlinkHref="#add"
          />
        </svg>
        <span>
          My list
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Featured film card correctly renders when needVanish not passed renders without wrapper 1`] = `
Array [
  "",
  <div
    className="movie-card__desc"
  >
    <h2
      className="movie-card__title"
    >
      The Aftermath
    </h2>
    <p
      className="movie-card__meta"
    >
      <span
        className="movie-card__genre"
      >
        Drama, Romance, War
      </span>
      <span
        className="movie-card__year"
      >
        2019
      </span>
    </p>
    <div
      className="movie-card__buttons"
    >
      <button
        className="btn btn--play movie-card__button"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <button
        className="btn btn--list movie-card__button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          height="20"
          viewBox="0 0 19 20"
          width="19"
        >
          <use
            xlinkHref="#add"
          />
        </svg>
        <span>
          My list
        </span>
      </button>
    </div>
  </div>,
]
`;
